std/assembly/string.ts

```diff
@@ -188,12 +188,13 @@ import { Array } from "./array";
   // TODO: implement full locale comparison with locales and Collator options
   localeCompare(other: String): i32 {
     if (changetype<usize>(other) == changetype<usize>(this)) return 0;
-    var len: isize = this.length;
-    var otherLen: isize = other.length;
-    if (otherLen != len) return select(1, -1, len > otherLen);
-    if (!otherLen) return 0; // "" == ""
+    var alen = this.length;
+    var blen = other.length;
     // @ts-ignore: string <-> String
-    return compareImpl(this, 0, other, 0, otherLen);
+    var res = compareImpl(this, 0, other, 0, <usize>min(alen, blen));
```

Comment:
Wouldn't this now run `compareImpl` even if the strings have different length, which could otherwise be a fast-path?

Reply:
Basically, lexicographically compare require run full compare and check length afterwards. "Fast path" before was incorrect. Btw it was same mistake in sort's comparison but it was fixed some time ago.

Reply:
Simple example while fast path fails:
```ts
"abcd".localeCompare("abc") // -> 1
"abdd".localeCompare("abz") // -> -1   // before we returned +1
```